rbs/rbs_common.h

```diff
@@ -1,15 +1,57 @@
 #ifndef SORBET_RBS_COMMON_H
 #define SORBET_RBS_COMMON_H
 
+#include <type_traits>
+
 extern "C" {
 #include "include/rbs.h"
 }
 
 #include "core/LocOffsets.h"
 #include "rbs/Parser.h"
 
+// Macro to check if an RBS location range is null (unset)
+// From ext/rbs_extension/legacy_location.c in the RBS repository
+#define NULL_LOC_RANGE_P(rg) ((rg).start == -1)
+
 namespace sorbet::rbs {
 
+/**
+ * Creates an rbs_string_t from a std::string_view.
+ *
+ * The returned rbs_string_t contains pointers into str's buffer, so str_view must outlive
+ * the rbs_string_t and any objects that use it.
+ */
+rbs_string_t makeRBSString(const std::string_view str);
+
+// Helper template to map RBS node types to their type enum values
+template <typename Type> struct RBSNodeTypeHelper {};
+
+#define DEF_RBS_TYPE_HELPER(rbs_struct_type, type_enum)         \
+    template <> struct RBSNodeTypeHelper<rbs_struct_type> {     \
+        static constexpr enum rbs_node_type TypeID = type_enum; \
+    }
+
+DEF_RBS_TYPE_HELPER(rbs_ast_symbol_t, RBS_AST_SYMBOL);
+DEF_RBS_TYPE_HELPER(rbs_ast_type_param_t, RBS_AST_TYPE_PARAM);
+DEF_RBS_TYPE_HELPER(rbs_types_function_t, RBS_TYPES_FUNCTION);
+DEF_RBS_TYPE_HELPER(rbs_types_function_param_t, RBS_TYPES_FUNCTION_PARAM);
+
+#undef DEF_RBS_TYPE_HELPER
+
+// Safe down_cast for RBS nodes (from rbs_node_t to specific types)
+// In debug builds, checks the node type before casting
+template <typename RBSNode> RBSNode *rbs_down_cast(rbs_node_t *anyNode) {
+    static_assert(std::is_same_v<decltype(RBSNode::base), rbs_node_t>,
```

Comment:
What types will this helper be called on? It's not used anywhere yet, and it wasn't clear to me what to expect for the `base` field for the types it's expected to be called on.  (I had a look at #9771, but didn't see any uses of `rbs_down_cast` there.)